### DISCUSSION TITLE
Support Keyboard Shortcuts for Buttons

### DISCUSSION THREAD

**Discussion by idvorkin** [hyperdiv/hyperdiv#72] [Ideas]
I love hyperdiv. It's sooo easy to make python web UX! 

It'd be great to have keyboard support with some kind of on_keypress.  
My use case is I want to be able to use the keyboard to press the buttons  when the input dialog doesn't have focus.

<img width="354" alt="image" src="https://github.com/user-attachments/assets/fa1e3f64-2f93-40b3-9c43-82ae8e10f56f">

**Comment by idvorkin** [comment]
OK, I got it working by making a quick and dirty [MouseTrap](https://craig.is/killing/mice#api.bind) plugin. @mariusnita, have you thought of the right implementation for keyboard input, if it's easy enough I might be able to create a PR? 

My quick and dirty plugin: 
https://github.com/idvorkin/mathflash/commit/2299b41a54d1b57adea6cf5654e7a96c89767390

**Reply by mariusnita** [reply]
I really like your simple solution. I like that keyboard events will fire only when the plugin is used.

I've avoided adding all the possible UI events to Hyperdiv (also missing are most mouse events) because they might cause a bunch of websocket chatter, especially mouseover. I have ideas for making a subscription system, so that only the events that are used in the app are actually fired by the UI, but it's not totally trivial to do.
